Won't build on Ubuntu 16, CUDA 9.0

I've tried to build TF from source, a number of times over the span of a week, each day pulling the latest TF.  I've tried gcc and I've tried clang.  With gcc it fails near the end.  If needed I'll open a separate issue for clang (the error is very different)

First I'll give configuration, then the FAIL output.

Configuration
--------------
 - bazel 0.11.1
 - python3.5 (/home/ubuntu/envs/med/lib/python3.5/site-packages)
 - jemalloc
 - No Google Cloud Platform support
 - No Hadoop File System support
 - Amazon S3 File System support enabled
 - No Apache Kafka Platform support
 - No XLA JIT support
 - No GDR support
 - No VERBS support
 - No OpenCL SYCL support
 - CUDA support
 - CUDA SDK 9.0
 - cuDNN version 7.0
 - TensorRT support will be enabled
 - NCCL 1.3
 - Nvidia compute capability 7.0
 - nvcc will be used as CUDA compiler
 - MPI support will be enabled for TensorFlow.
 - optimization flags during compilation when bazel option "--config=opt": -march=native
 - Not configuring the WORKSPACE for Android builds.


Latest fail output:
-----------------
... (many warnings before this)

./tensorflow/core/kernels/cwise_ops.h(199): warning: __device__ annotation on a defaulted function("scalar_right") is ignored

./tensorflow/core/kernels/cwise_ops.h(169): warning: __host__ annotation on a defaulted function("scalar_left") is ignored

./tensorflow/core/kernels/cwise_ops.h(169): warning: __device__ annotation on a defaulted function("scalar_left") is ignored

./tensorflow/core/kernels/cwise_ops.h(199): warning: __host__ annotation on a defaulted function("scalar_right") is ignored

./tensorflow/core/kernels/cwise_ops.h(199): warning: __device__ annotation on a defaulted function("scalar_right") is ignored

INFO: From Compiling tensorflow/core/kernels/data/repeat_dataset_op.cc:
tensorflow/core/kernels/data/repeat_dataset_op.cc: In member function 'virtual void tensorflow::{anonymous}::RepeatDatasetOp::MakeDataset(tensorflow::OpKernelContext*, tensorflow::DatasetBase*, tensorflow::DatasetBase**)':
tensorflow/core/kernels/data/repeat_dataset_op.cc:45:61: warning: 'count' may be used uninitialized in this function [-Wmaybe-uninitialized]
         : GraphDatasetBase(ctx), count_(count), input_(input) {
                                                             ^
tensorflow/core/kernels/data/repeat_dataset_op.cc:36:11: note: 'count' was declared here
     int64 count;
           ^
INFO: From Compiling tensorflow/core/kernels/cwise_op_gpu_bitwise_and.cu.cc:
./tensorflow/core/kernels/cwise_ops.h(169): warning: __host__ annotation on a defaulted function("scalar_left") is ignored

./tensorflow/core/kernels/cwise_ops.h(169): warning: __device__ annotation on a defaulted function("scalar_left") is ignored

./tensorflow/core/kernels/cwise_ops.h(199): warning: __host__ annotation on a defaulted function("scalar_right") is ignored

./tensorflow/core/kernels/cwise_ops.h(199): warning: __device__ annotation on a defaulted function("scalar_right") is ignored

./tensorflow/core/kernels/cwise_ops.h(169): warning: __host__ annotation on a defaulted function("scalar_left") is ignored

./tensorflow/core/kernels/cwise_ops.h(169): warning: __device__ annotation on a defaulted function("scalar_left") is ignored

./tensorflow/core/kernels/cwise_ops.h(199): warning: __host__ annotation on a defaulted function("scalar_right") is ignored

./tensorflow/core/kernels/cwise_ops.h(199): warning: __device__ annotation on a defaulted function("scalar_right") is ignored

INFO: From ProtoCompile tensorflow/core/debug/debugger_event_metadata.pb.cc:
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
INFO: From ProtoCompile tensorflow/contrib/mpi/mpi_msg.pb.cc:
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
INFO: From ProtoCompile tensorflow/python/framework/cpp_shape_inference.pb.cc:
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
INFO: From ProtoCompile tensorflow/core/debug/debug_service.pb.cc:
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
bazel-out/k8-py3-opt/genfiles/external/protobuf_archive/src: warning: directory does not exist.
INFO: From Compiling tensorflow/core/grappler/optimizers/graph_optimizer_stage.cc:
tensorflow/core/grappler/optimizers/graph_optimizer_stage.cc: In function 'tensorflow::Status tensorflow::grappler::GetTensorProperties(const tensorflow::grappler::GraphOptimizerContext&, const string&, tensorflow::OpInfo::TensorProperties*)':
tensorflow/core/grappler/optimizers/graph_optimizer_stage.cc:60:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   if (num_outputs == 0 || port > num_outputs - 1) {
                                ^
ERROR: /home/ubuntu/downloads/tensorflow/tensorflow/contrib/mpi/BUILD:60:1: C++ compilation of rule '//tensorflow/contrib/mpi:mpi_rendezvous_mgr' failed (Exit 1)
In file included from ./tensorflow/contrib/mpi/mpi_rendezvous_mgr.h:34:0,
                 from tensorflow/contrib/mpi/mpi_rendezvous_mgr.cc:18:
./tensorflow/contrib/mpi/mpi_utils.h: In member function 'const int tensorflow::MPIUtils::GetSourceID(const string&) const':
./tensorflow/contrib/mpi/mpi_utils.h:49:11: error: 'FATAL' was not declared in this scope
       LOG(FATAL) << "Failed to convert worker name to MPI index: " << task_id;
           ^
./tensorflow/contrib/mpi/mpi_utils.h:49:16: error: 'LOG' was not declared in this scope
       LOG(FATAL) << "Failed to convert worker name to MPI index: " << task_id;
                ^
In file included from ./tensorflow/core/lib/strings/strcat.h:27:0,
                 from ./tensorflow/core/lib/strings/str_util.h:23,
                 from ./tensorflow/contrib/mpi/mpi_utils.h:25,
                 from ./tensorflow/contrib/mpi/mpi_rendezvous_mgr.h:34,
                 from tensorflow/contrib/mpi/mpi_rendezvous_mgr.cc:18:
./tensorflow/core/util/tensor_format.h: In instantiation of 'T tensorflow::GetTensorDim(tensorflow::gtl::ArraySlice<T>, tensorflow::TensorFormat, char) [with T = long long int]':
./tensorflow/core/util/tensor_format.h:372:47:   required from here
./tensorflow/core/util/tensor_format.h:340:29: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   CHECK(index >= 0 && index < dimension_attributes.size())
                             ^
./tensorflow/core/platform/macros.h:82:47: note: in definition of macro 'TF_PREDICT_FALSE'
 #define TF_PREDICT_FALSE(x) (__builtin_expect(x, 0))
                                               ^
./tensorflow/core/util/tensor_format.h:340:3: note: in expansion of macro 'CHECK'
   CHECK(index >= 0 && index < dimension_attributes.size())
   ^
./tensorflow/core/util/tensor_format.h: In instantiation of 'T tensorflow::GetFilterDim(tensorflow::gtl::ArraySlice<T>, tensorflow::FilterTensorFormat, char) [with T = long long int]':
./tensorflow/core/util/tensor_format.h:381:54:   required from here
./tensorflow/core/util/tensor_format.h:355:29: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   CHECK(index >= 0 && index < dimension_attribute.size())
                             ^
./tensorflow/core/platform/macros.h:82:47: note: in definition of macro 'TF_PREDICT_FALSE'
 #define TF_PREDICT_FALSE(x) (__builtin_expect(x, 0))
                                               ^
./tensorflow/core/util/tensor_format.h:355:3: note: in expansion of macro 'CHECK'
   CHECK(index >= 0 && index < dimension_attribute.size())
   ^
Target //tensorflow/tools/pip_package:build_pip_package failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 2688.392s, Critical Path: 200.30s
FAILED: Build did NOT complete successfully

Incidentally, here's the FAIL output when using clang:

uested here
    return numext::numeric_limits<T>::quiet_NaN();
                                      ^
./tensorflow/core/kernels/reduction_ops.h:59:1: note: in instantiation of member function 'Eigen::GenericNumTraits<std::complex<float> >::quiet_NaN' requested here
FIX_MEAN_IDENTITY(complex64)
^
./tensorflow/core/kernels/reduction_ops.h:53:35: note: expanded from macro 'FIX_MEAN_IDENTITY'
      return Eigen::NumTraits<T>::quiet_NaN();                  \
                                  ^
In file included from tensorflow/core/kernels/reduction_ops_gpu_bool.cu.cc:20:
In file included from ./tensorflow/core/kernels/reduction_gpu_kernels.cu.h:20:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:14:
In file included from external/eigen_archive/Eigen/Core:382:
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/Meta.h:187:84: warning: control reaches end of non-void function [-Wreturn-type]
  static T quiet_NaN() { assert(false && "quiet_NaN not supported for this type"); }
                                                                                   ^
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/NumTraits.h:180:39: note: in instantiation of member function 'Eigen::internal::device::numeric_limits<std::complex<double> >::quiet_NaN' requested here
    return numext::numeric_limits<T>::quiet_NaN();
                                      ^
./tensorflow/core/kernels/reduction_ops.h:60:1: note: in instantiation of member function 'Eigen::GenericNumTraits<std::complex<double> >::quiet_NaN' requested here
FIX_MEAN_IDENTITY(complex128)
^
./tensorflow/core/kernels/reduction_ops.h:53:35: note: expanded from macro 'FIX_MEAN_IDENTITY'
      return Eigen::NumTraits<T>::quiet_NaN();                  \
                                  ^
2 warnings generated when compiling for sm_70.
INFO: From Compiling tensorflow/core/kernels/reduction_ops_half_prod_max_min.cu.cc:
In file included from tensorflow/core/kernels/reduction_ops_half_prod_max_min.cu.cc:20:
In file included from ./tensorflow/core/kernels/reduction_gpu_kernels.cu.h:20:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:14:
In file included from external/eigen_archive/Eigen/Core:382:
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/Meta.h:187:84: warning: control reaches end of non-void function [-Wreturn-type]
  static T quiet_NaN() { assert(false && "quiet_NaN not supported for this type"); }
                                                                                   ^
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/NumTraits.h:180:39: note: in instantiation of member function 'Eigen::internal::device::numeric_limits<std::complex<float> >::quiet_NaN' requested here
    return numext::numeric_limits<T>::quiet_NaN();
                                      ^
./tensorflow/core/kernels/reduction_ops.h:59:1: note: in instantiation of member function 'Eigen::GenericNumTraits<std::complex<float> >::quiet_NaN' requested here
FIX_MEAN_IDENTITY(complex64)
^
./tensorflow/core/kernels/reduction_ops.h:53:35: note: expanded from macro 'FIX_MEAN_IDENTITY'
      return Eigen::NumTraits<T>::quiet_NaN();                  \
                                  ^
In file included from tensorflow/core/kernels/reduction_ops_half_prod_max_min.cu.cc:20:
In file included from ./tensorflow/core/kernels/reduction_gpu_kernels.cu.h:20:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:14:
In file included from external/eigen_archive/Eigen/Core:382:
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/Meta.h:187:84: warning: control reaches end of non-void function [-Wreturn-type]
  static T quiet_NaN() { assert(false && "quiet_NaN not supported for this type"); }
                                                                                   ^
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/NumTraits.h:180:39: note: in instantiation of member function 'Eigen::internal::device::numeric_limits<std::complex<double> >::quiet_NaN' requested here
    return numext::numeric_limits<T>::quiet_NaN();
                                      ^
./tensorflow/core/kernels/reduction_ops.h:60:1: note: in instantiation of member function 'Eigen::GenericNumTraits<std::complex<double> >::quiet_NaN' requested here
FIX_MEAN_IDENTITY(complex128)
^
./tensorflow/core/kernels/reduction_ops.h:53:35: note: expanded from macro 'FIX_MEAN_IDENTITY'
      return Eigen::NumTraits<T>::quiet_NaN();                  \
                                  ^
2 warnings generated when compiling for sm_70.
INFO: From Compiling tensorflow/core/kernels/softmax_op.cc:
In file included from tensorflow/core/kernels/softmax_op.cc:26:
./tensorflow/core/kernels/softmax_op_functor.h:60:45: warning: unused variable 'depth_dim' [-Wunused-variable]
    Eigen::IndexList<Eigen::type2index<1> > depth_dim;
                                            ^
1 warning generated.
INFO: From Compiling tensorflow/core/kernels/l2loss_op_gpu.cu.cc:
In file included from tensorflow/core/kernels/l2loss_op_gpu.cu.cc:20:
In file included from ./tensorflow/core/kernels/l2loss_op.h:18:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:14:
In file included from external/eigen_archive/Eigen/Core:382:
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/Meta.h:187:84: warning: control reaches end of non-void function [-Wreturn-type]
  static T quiet_NaN() { assert(false && "quiet_NaN not supported for this type"); }
                                                                                   ^
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/NumTraits.h:180:39: note: in instantiation of member function 'Eigen::internal::device::numeric_limits<std::complex<float> >::quiet_NaN' requested here
    return numext::numeric_limits<T>::quiet_NaN();
                                      ^
./tensorflow/core/kernels/reduction_ops.h:59:1: note: in instantiation of member function 'Eigen::GenericNumTraits<std::complex<float> >::quiet_NaN' requested here
FIX_MEAN_IDENTITY(complex64)
^
./tensorflow/core/kernels/reduction_ops.h:53:35: note: expanded from macro 'FIX_MEAN_IDENTITY'
      return Eigen::NumTraits<T>::quiet_NaN();                  \
                                  ^
In file included from tensorflow/core/kernels/l2loss_op_gpu.cu.cc:20:
In file included from ./tensorflow/core/kernels/l2loss_op.h:18:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:14:
In file included from external/eigen_archive/Eigen/Core:382:
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/Meta.h:187:84: warning: control reaches end of non-void function [-Wreturn-type]
  static T quiet_NaN() { assert(false && "quiet_NaN not supported for this type"); }
                                                                                   ^
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/NumTraits.h:180:39: note: in instantiation of member function 'Eigen::internal::device::numeric_limits<std::complex<double> >::quiet_NaN' requested here
    return numext::numeric_limits<T>::quiet_NaN();
                                      ^
./tensorflow/core/kernels/reduction_ops.h:60:1: note: in instantiation of member function 'Eigen::GenericNumTraits<std::complex<double> >::quiet_NaN' requested here
FIX_MEAN_IDENTITY(complex128)
^
./tensorflow/core/kernels/reduction_ops.h:53:35: note: expanded from macro 'FIX_MEAN_IDENTITY'
      return Eigen::NumTraits<T>::quiet_NaN();                  \
                                  ^
2 warnings generated when compiling for sm_70.
INFO: From Compiling tensorflow/core/kernels/softmax_op_gpu.cu.cc:
In file included from tensorflow/core/kernels/softmax_op_gpu.cu.cc:21:
In file included from ./tensorflow/core/framework/op_kernel.h:23:
In file included from ./tensorflow/core/framework/allocator.h:23:
In file included from ./tensorflow/core/framework/numeric_types.h:20:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:14:
In file included from external/eigen_archive/Eigen/Core:382:
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/Meta.h:187:84: warning: control reaches end of non-void function [-Wreturn-type]
  static T quiet_NaN() { assert(false && "quiet_NaN not supported for this type"); }
                                                                                   ^
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/NumTraits.h:180:39: note: in instantiation of member function 'Eigen::internal::device::numeric_limits<std::complex<float> >::quiet_NaN' requested here
    return numext::numeric_limits<T>::quiet_NaN();
                                      ^
./tensorflow/core/kernels/reduction_ops.h:59:1: note: in instantiation of member function 'Eigen::GenericNumTraits<std::complex<float> >::quiet_NaN' requested here
FIX_MEAN_IDENTITY(complex64)
^
./tensorflow/core/kernels/reduction_ops.h:53:35: note: expanded from macro 'FIX_MEAN_IDENTITY'
      return Eigen::NumTraits<T>::quiet_NaN();                  \
                                  ^
In file included from tensorflow/core/kernels/softmax_op_gpu.cu.cc:21:
In file included from ./tensorflow/core/framework/op_kernel.h:23:
In file included from ./tensorflow/core/framework/allocator.h:23:
In file included from ./tensorflow/core/framework/numeric_types.h:20:
In file included from ./third_party/eigen3/unsupported/Eigen/CXX11/Tensor:1:
In file included from external/eigen_archive/unsupported/Eigen/CXX11/Tensor:14:
In file included from external/eigen_archive/Eigen/Core:382:
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/util/Meta.h:187:84: warning: control reaches end of non-void function [-Wreturn-type]
  static T quiet_NaN() { assert(false && "quiet_NaN not supported for this type"); }
                                                                                   ^
external/eigen_archive/unsupported/Eigen/CXX11/../../../Eigen/src/Core/NumTraits.h:180:39: note: in instantiation of member function 'Eigen::internal::device::numeric_limits<std::complex<double> >::quiet_NaN' requested here
    return numext::numeric_limits<T>::quiet_NaN();
                                      ^
./tensorflow/core/kernels/reduction_ops.h:60:1: note: in instantiation of member function 'Eigen::GenericNumTraits<std::complex<double> >::quiet_NaN' requested here
FIX_MEAN_IDENTITY(complex128)
^
./tensorflow/core/kernels/reduction_ops.h:53:35: note: expanded from macro 'FIX_MEAN_IDENTITY'
      return Eigen::NumTraits<T>::quiet_NaN();                  \
                                  ^
2 warnings generated when compiling for sm_70.
INFO: From Compiling tensorflow/contrib/lite/toco/toco.cc:
In file included from tensorflow/contrib/lite/toco/toco.cc:20:
In file included from ./tensorflow/contrib/lite/toco/model.h:26:
In file included from ./tensorflow/contrib/lite/toco/runtime/types.h:18:
In file included from ./tensorflow/contrib/lite/kernels/internal/common.h:42:
external/arm_neon_2_x86_sse/NEON_2_SSE.h:4750:27: warning: implicit conversion from 'int' to 'char' changes value from 128 to -128 [-Wconstant-conversion]
    c128 = _mm_set1_epi8 (128);
           ~~~~~~~~~~~~~  ^~~
external/arm_neon_2_x86_sse/NEON_2_SSE.h:5431:27: warning: implicit conversion from 'int' to 'char' changes value from 128 to -128 [-Wconstant-conversion]
    c128 = _mm_set1_epi8 (128);
           ~~~~~~~~~~~~~  ^~~
2 warnings generated.
ERROR: /home/ubuntu/downloads/tensorflow/tensorflow/contrib/mpi_collectives/BUILD:40:1: C++ compilation of rule '//tensorflow/contrib/mpi_collectives:python/ops/_mpi_ops.so' failed (Exit 1)
tensorflow/contrib/mpi_collectives/kernels/ring.cc:58:6: error: explicit specialization of 'CopyTensorData<Eigen::ThreadPoolDevice>' after instantiation
void CopyTensorData<CPUDevice>(void* dst, void* src, size_t size) {
     ^
./tensorflow/contrib/mpi_collectives/kernels/ring.h:165:3: note: implicit instantiation first required here
  CopyTensorData<Device>((void*)buffer, (void*)input->tensor_data().data(),
  ^
tensorflow/contrib/mpi_collectives/kernels/ring.cc:71:1: error: explicit specialization of 'AccumulateTensorData<Eigen::ThreadPoolDevice, int>' after instantiation
GENERATE_ACCUMULATE(int);
^
tensorflow/contrib/mpi_collectives/kernels/ring.cc:65:8: note: expanded from macro 'GENERATE_ACCUMULATE'
  void AccumulateTensorData<CPUDevice, type>(type * dst, type * src, \
       ^
./tensorflow/contrib/mpi_collectives/kernels/ring.h:221:5: note: implicit instantiation first required here
    AccumulateTensorData<Device, T>(segment_update, segment_recv,
    ^
tensorflow/contrib/mpi_collectives/kernels/ring.cc:72:1: error: explicit specialization of 'AccumulateTensorData<Eigen::ThreadPoolDevice, long long>' after instantiation
GENERATE_ACCUMULATE(long long);
^
tensorflow/contrib/mpi_collectives/kernels/ring.cc:65:8: note: expanded from macro 'GENERATE_ACCUMULATE'
  void AccumulateTensorData<CPUDevice, type>(type * dst, type * src, \
       ^
./tensorflow/contrib/mpi_collectives/kernels/ring.h:221:5: note: implicit instantiation first required here
    AccumulateTensorData<Device, T>(segment_update, segment_recv,
    ^
tensorflow/contrib/mpi_collectives/kernels/ring.cc:73:1: error: explicit specialization of 'AccumulateTensorData<Eigen::ThreadPoolDevice, float>' after instantiation
GENERATE_ACCUMULATE(float);
^
tensorflow/contrib/mpi_collectives/kernels/ring.cc:65:8: note: expanded from macro 'GENERATE_ACCUMULATE'
  void AccumulateTensorData<CPUDevice, type>(type * dst, type * src, \
       ^
./tensorflow/contrib/mpi_collectives/kernels/ring.h:221:5: note: implicit instantiation first required here
    AccumulateTensorData<Device, T>(segment_update, segment_recv,
    ^
4 errors generated.
Target //tensorflow/tools/pip_package:build_pip_package failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 2303.291s, Critical Path: 108.87s
FAILED: Build did NOT complete successfully

--- a/tensorflow/contrib/mpi/mpi_utils.h
+++ b/tensorflow/contrib/mpi/mpi_utils.h
@@ -22,6 +22,7 @@ limitations under the License.
 #include <string>
 #include <vector>
 
+#include "tensorflow/core/platform/logging.h"
 #include "tensorflow/core/lib/strings/str_util.h"
 
 // Skip MPI C++ bindings support, this matches the usage in other places

